### PR TITLE
IGNITE-13982 Add documentation for new checkpoint, cluster and cache metrics

### PR DESCRIPTION
**Pull request by NSAmelchev**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 1]
```diff
@@ -78,11 +79,16 @@ Register name: `cache.{cache_name}.{near}`
 |EstimatedRebalancingKeys|long|Number estimated to rebalance keys.
 |GetTime |histogram|   Get time in nanoseconds.
 |GetTimeTotal|long|The total time of cache gets, in nanoseconds.
+|HeapEntriesCount|long|Onheap entries count.
+|IndexRebuildKeyProcessed|long|Number of keys processed during index rebuilding.
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:83]:
```suggestion
|IndexRebuildKeyProcessed|long|Number of keys processed during the index rebuilding.
```

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 2]
```diff
@@ -340,3 +364,21 @@ Register name: `io.datastorage`
...
+Cluster metrics.
+
+Register name: `cluster`
+
+
+[cols="2,1,3",opts="header"]
+|===
+|Name|    Type|    Description
+|ActiveBaselineNodes| integer | Active baseline nodes count.
+|Rebalanced| boolean | True if the cluster has achieved fully rebalanced state. Note that an inactive cluster always has this metric in False regardless of the real partitions state.
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:380]:
```suggestion
|Rebalanced| boolean | True if the cluster has fully achieved rebalanced state. Note that an inactive cluster always has this metric in False regardless of the real partitions state.
```

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 3]
```diff
@@ -320,15 +326,33 @@ Register name: `io.datastorage`
...
+|LastCheckpointListenersExecuteDuration|  long|    Duration of the checkpoint execution listeners under write lock in milliseconds.
+|LastCheckpointLockHoldDuration|  long|    Duration of the checkpoint lock hold in milliseconds.
 |LastCheckpointLockWaitDuration|  long|    Duration of the checkpoint lock wait in milliseconds.
-|LastCheckpointMarkDuration | long  |  Duration of the checkpoint lock wait in milliseconds.
+|LastCheckpointMarkDuration | long  |  Duration of the checkpoint mark in milliseconds.
 |LastCheckpointPagesWriteDuration|    long|    Duration of the checkpoint pages write in milliseconds.
 |LastCheckpointTotalPagesNumber|  long|    Total number of pages written during the last checkpoint.
+|LastCheckpointSplitAndSortPagesDuration|  long|    Duration of splitting and sorting checkpoint pages of the last checkpoint in milliseconds.
+|LastCheckpointStart|  long|    Start timestamp of the last checkpoint.
+|LastCheckpointWalRecordFsyncDuration|  long|    Duration of the WAL fsync after logging CheckpointRecord on begin of the last checkpoint in milliseconds.
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:354]:
```suggestion
|LastCheckpointWalRecordFsyncDuration|  long|    Duration of the WAL fsync after logging CheckpointRecord on the start of the last checkpoint in milliseconds.
```

**Comment by NSAmelchev** [comment]
@Nikita-tech-writer , thank you for the review. I have fixed issues.

File: docs/_docs/monitoring-metrics/new-metrics.adoc (hunk 3 above)
Comment (Nikita-tech-writer) [reply docs/_docs/monitoring-metrics/new-metrics.adoc]: or "...at the beginning of the last checkpoint in milliseconds."

File: docs/_docs/monitoring-metrics/new-metrics.adoc [hunk 4]
```diff
@@ -320,15 +326,33 @@ Register name: `io.datastorage`
...
+|CheckpointHistogram| histogram |   Histogram of checkpoint durations in milliseconds.
+|CheckpointListenersExecuteHistogram| histogram |   Histogram of checkpoint execution listeners under write lock durations in milliseconds.
+|CheckpointLockHoldHistogram| histogram |   Histogram of checkpoint lock hold durations in milliseconds.
+|CheckpointLockWaitHistogram| histogram |   Histogram of checkpoint lock wait durations in milliseconds.
+|CheckpointMarkHistogram| histogram |   Histogram of checkpoint mark durations in milliseconds.
+|CheckpointPagesWriteHistogram| histogram |   Histogram of checkpoint pages write durations in milliseconds.
+|CheckpointSplitAndSortPagesHistogram| histogram |   Histogram of splitting and sorting checkpoint pages durations in milliseconds.
 |CheckpointTotalTime| long |   Total duration of checkpoint
+|CheckpointWalRecordFsyncHistogram| histogram |   Histogram of the WAL fsync after logging ChTotalNodeseckpointRecord on begin of checkpoint durations in milliseconds.
+|CheckpointWriteEntryHistogram| histogram |   Histogram of entry buffer writing to file durations in milliseconds.
```
Comment (Nikita-tech-writer) [inline docs/_docs/monitoring-metrics/new-metrics.adoc:340]: Are you sure that it is "duration**s**" (plural), not "duration" (singular) here?

**Review by Nikita-tech-writer — CHANGES_REQUESTED** [review]
Please see my comments to the PR.

File: docs/_docs/monitoring-metrics/new-metrics.adoc (hunk 4 above)
Comment (NSAmelchev) [reply docs/_docs/monitoring-metrics/new-metrics.adoc]: @Nikita-tech-writer , It seems that singular is correct. I have fixed docs and metrics descriptions.

**Comment by NSAmelchev** [comment]
@Nikita-tech-writer I have fixed issues.

**Review by Mmuzaf — APPROVED** [review]

**Pull request merged**